config/utility.py

```diff
@@ -18,6 +18,20 @@ def AddSystematicShift(
         producer[0].shift(name, producer[1])
 
 
+# Function for introducing systematic variations to producers and depending quantities by adding an already shifted input quantity
+def SystematicShiftByInputQuantity(
+    config, name, base_producers, external_dict, sanetize_producers=[]
+):
+    for scope in external_dict.keys():
+        for quantity in external_dict[scope].keys():
+            quantity.register_external_shift(
+                shift="__" + name,
+                scope=scope,
+                external_name=external_dict[scope][quantity],
+            )
+    AddSystematicShift(config, name, {}, base_producers, sanetize_producers)
```

Comment:
I think, there is no need for the call of `AddSystematicShift`. Also, there is no need for scope treatment (see comment above). You only need to call register_external_shift with shiftname and quantityname. If function above is adapted correspondingly this will propagate through all dependent quantities. This is independent from producers, which were only used beforehand to create dependencies between quantities.
Sidemark: In the AddSystematicShift method, the producers are used because systematic variations are generated by changing their parameters and behavior such that there is a direct logic link between variation and producer (this is basically userfriendliness). For the propagation it would also work to directly shift their output quantities.

Reply:
quantity class changes are reversed